firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandler.java

```diff
@@ -189,21 +189,32 @@ public ConfigFetchHandler(
                   createThrottledMessage(backoffEndTime.getTime() - currentTime.getTime()),
                   backoffEndTime.getTime()));
     } else {
-      Task<InstanceIdResult> instanceIdTask = firebaseInstanceId.getInstanceId();
+      Task<String> installationIdTask = firebaseInstallations.getId();
+      Task<InstallationTokenResult> installationTokenTask = firebaseInstallations.getToken(false);
       fetchResponseTask =
-          instanceIdTask.continueWithTask(
-              executor,
-              (completedIidTask) -> {
-                if (!completedIidTask.isSuccessful()) {
-                  return Tasks.forException(
-                      new FirebaseRemoteConfigClientException(
-                          "Failed to get Firebase Instance ID token for fetch.",
-                          completedIidTask.getException()));
-                }
-
-                InstanceIdResult instanceIdResult = completedIidTask.getResult();
-                return fetchFromBackendAndCacheResponse(instanceIdResult, currentTime);
-              });
+          Tasks.whenAllComplete(installationIdTask, installationTokenTask)
+              .continueWithTask(
+                  executor,
+                  (completedInstallationsTasks) -> {
+                    if (!installationIdTask.isSuccessful()) {
+                      return Tasks.forException(
+                          new FirebaseRemoteConfigClientException(
+                              "Failed to get Firebase Installation ID for fetch.",
```

Comment:
How about mentioning "`Firebase Installations`" so that developers have all the info to potentially raise issues with Firebase?

Reply:
Good call